docs/CONTEXT.md

```diff
@@ -0,0 +1,96 @@
+# Context object
+`HullApp` applies multiple middleware to the request handler.
+The result is `req.hull` object which is the Context Object - a set of modules to work in the context of current organization and connector instance.
+
+```javascript
+{
+  client: {
+    logger: {},
+  },
+  ship: {},
+  token: "",
+  segments: [],
+  hostname: req.hostname,
+  cache: {},
+  enqueue: () => {},
+  metric: {},
+  service: {},
+  message: {},
+  notification: {}
+}
+```
+
+#### client
+[Hull API client](../README.md) initialized to work with current organization
+
+#### ship
+Ship object with manifest information and `private_settings`
+
+#### token
+A string with encoded credentials
+
+#### segments
+An array of segments information
+
+#### hostname
+Hostname of the current request.
+
+#### cache
+[Caching object](../src/infra/cache/ship-cache.js)
```

Comment:
need example of how it's used and why it's exposed. since we already have the `ship` object exposed.